server/sonar-server/src/main/java/org/sonar/server/permission/index/PermissionIndexer.java

```diff
@@ -58,153 +56,137 @@
         bulkRequest = esClient.prepareBulk().setRefresh(false);
         count = 0;
       }
     }
-    EsUtils.executeBulkRequest(bulkRequest, BULK_ERROR_MESSAGE);
-    esClient.prepareRefresh(IssueIndexDefinition.INDEX).get();
-    esClient.prepareRefresh(ProjectMeasuresIndexDefinition.INDEX_PROJECT_MEASURES).get();
-    esClient.prepareRefresh(ComponentIndexDefinition.INDEX_COMPONENTS).get();
-  }
-
-  public void index(DbSession dbSession, String viewOrProjectUuid) {
-    PermissionIndexerDao dao = new PermissionIndexerDao();
-    List<PermissionIndexerDao.Dto> dtos = dao.selectByUuids(dbClient, dbSession, singletonList(viewOrProjectUuid));
-    if (dtos.size() == 1) {
-      index(dtos.get(0));
-    }
-  }
-
-  @VisibleForTesting
-  void index(PermissionIndexerDao.Dto dto) {
-    newIssuesAuthorizationIndexRequest(dto)
-      .ifPresent(rqst -> index(IssueIndexDefinition.INDEX, IssueIndexDefinition.TYPE_AUTHORIZATION, rqst));
-    newProjectMeasuresAuthorizationIndexRequest(dto)
-      .ifPresent(rqst -> index(ProjectMeasuresIndexDefinition.INDEX_PROJECT_MEASURES, ProjectMeasuresIndexDefinition.TYPE_AUTHORIZATION, rqst));
-    newComponentsAuthorizationIndexRequest(dto)
-      .ifPresent(rqst -> index(ComponentIndexDefinition.INDEX_COMPONENTS, ComponentIndexDefinition.TYPE_AUTHORIZATION, rqst));
-  }
-
-  private void index(String index, String type, IndexRequest indexRequest) {
-    esClient.prepareIndex(index, type)
-      .setId(indexRequest.id())
-      .setRouting(indexRequest.routing())
-      .setSource(indexRequest.source())
-      .setRefresh(true)
-      .get();
-  }
-
-  private static Optional<IndexRequest> newIssuesAuthorizationIndexRequest(PermissionIndexerDao.Dto dto) {
-    if (!isProject(dto)) {
-      return Optional.empty();
+    if (count > 0) {
+      EsUtils.executeBulkRequest(bulkRequest, BULK_ERROR_MESSAGE);
     }
-    Map<String, Object> doc = ImmutableMap.of(
-      IssueIndexDefinition.FIELD_AUTHORIZATION_PROJECT_UUID, dto.getProjectUuid(),
-      IssueIndexDefinition.FIELD_AUTHORIZATION_GROUPS, dto.getGroups(),
-      IssueIndexDefinition.FIELD_AUTHORIZATION_USERS, dto.getUsers(),
-      IssueIndexDefinition.FIELD_AUTHORIZATION_UPDATED_AT, new Date(dto.getUpdatedAt()));
-    return Optional.of(
-      new IndexRequest(IssueIndexDefinition.INDEX, IssueIndexDefinition.TYPE_AUTHORIZATION, dto.getProjectUuid())
-        .routing(dto.getProjectUuid())
-        .source(doc));
+    authorizationScopes.forEach(type -> esClient.prepareRefresh(type.getIndexName()).get());
   }
 
-  private static Optional<IndexRequest> newProjectMeasuresAuthorizationIndexRequest(PermissionIndexerDao.Dto dto) {
-    if (!isProject(dto)) {
-      return Optional.empty();
+  private static IndexRequest newIndexRequest(PermissionIndexerDao.Dto dto, String indexName) {
+    Map<String, Object> doc = new HashMap<>();
+    doc.put(AuthorizationTypeSupport.FIELD_UPDATED_AT, new Date(dto.getUpdatedAt()));
```

Comment:
Ok, done. DateUtils is convenient when the Long parameter can be null. Else the regular new Date(long) is fine for me.